v1/formats/nswpdc7and8.xml

```diff
@@ -0,0 +1,57 @@
+<?xml version='1.0' encoding='UTF-8'?>
+<debate-format schema-version="2.2">
+  <name>NSW Premier's Debating Challenge (7 and 8)</name>
+  <short-name>NSWPDC7and8</short-name>
+  <version>1</version>
+  <info>
+    <region>NSW, Australia</region>
+    <level>Primary School</level>
+    <used-at>NSW Premier's Debating Challenge (for Years 7 and 8)</used-at>
+    <description>3 speakers + 4th vs 3 speakers + 4th, 6 minute speeches</description>
+  </info>
+  <period-types>
+   <period-type ref="pdc.1bell">
+     <name>PDC 1 Bell</name>
+     <display>1 Bell</display>
+     <default-bgcolor>#CCAA00</default-bgcolor>
+   </period-type>
+   <period-type ref="pdc.2bell">
+     <name>PDC 2 Bells</name>
+     <display>2 Bells</display>
+     <default-bgcolor>#CC6600</default-bgcolor>
+   </period-type>
+   <period-type ref="pdc.overtime">
+     <name>PDC Overtime</name>
+     <display>***OVERTIME***</display>
+     <default-bgcolor>#CC2200</default-bgcolor>
+   </period-type>
+  </period-types>
+  <prep-time length="60:00"/>
+  <speech-types>
+    <speech-type ref="speech" length="7:00" first-period="normal">
+      <bell time="4:00" number="1" next-period="pdc.1bell"/>
+      <bell time="6:00" number="2" next-period="pdc.2bell"/>
+      <bell time="finish" number="60" next-period="pdc.overtime"/>
```

Comment:
Unless there's a particular reason to use custom colours, I'd recommend just using the built-in period types, to be consistent with other formats.
```suggestion
      <bell time="4:00" number="1" next-period="warning"/>
      <bell time="6:00" number="2" next-period="overtime"/>
      <bell time="finish" number="60" next-period="overtime"/>
```
But this raises another discussion about the speech format, which I left in a separate comment on this pull request.

Reply:
I'd prefer to leave this as is (for tactical reasons). If you update the "grace period," so that it's editable within the format, I'll come back in and fix it up. Is that OK?

Reply:
Sorry I'm slow to come back to this—there are two issues that I kinda mangled together here, sorry: (1) the colours and period descriptions used, and (2) when the bells are set to ring. (2) is all good as it stands now, following our discussion. But for (1) I'd like to be a bit more consistent with other formats.

I've been trying to phase out the use of period descriptions in the normal case, leaving them only when the period demands description to keep track of what's happening. Examples ([here are all of them](https://github.com/search?q=repo%3Aczlee%2Fdebatekeeper-formats+period-type+path%3A%2F%5Ev1%5C%2Fformats%5C%2F%2F&type=code)):
- In NZ Officers' Cup, the first minute of prep time is used to choose a moot, and the second minute to choose a side, so those get labelled "Choose moot" and "Choose side"
- In Canadian formats the Leader of the Opposition can choose to speak last and speak for a single ten-minute block, rather than having separate substantive and rebuttal (what Oceania calls "reply") speeches. So at 7 minutes, the description "Rebuttal time" appears to indicate that rebuttal speech rules now apply.

As for colours, where warning bells are analogous I'm trying to use the built-in period types to keep colours consistent between formats. In this case I think we could use the existing `warning` (orange) and `overtime` (red) period types, but add a new `pdc.1bell` type to be yellow like you have? So something like

```suggestion
      <bell time="4:00" number="1" next-period="pdc.1bell"/>
      <bell time="6:00" number="2" next-period="warning"/>
      <bell time="finish" number="60" next-period="overtime"/>
```

Then only this period type is necessary (note no description):
``` xml
   <period-type ref="pdc.1bell">
     <name>PDC 1 Bell</name>
     <default-bgcolor>#CCAA00</default-bgcolor>
   </period-type>
```

Would that be sensible? I'll do this for all files and put it in a new branch/PR so you can see and try out the files.

(The default colours are here: https://github.com/czlee/debatekeeper/blob/master/app/src/main/assets/periods.xml)